include/abb_librws/rws_rapid.h

```diff
@@ -371,6 +372,243 @@ struct RAPIDRecord : public RAPIDSymbolDataAbstract
   std::vector<RAPIDSymbolDataAbstract*> components_;
 };
 
+/**
+ * \brief A struct, for representing the data of a RAPID array of atomic type.
+ */
+template<typename T>
+struct RAPIDArray : public RAPIDSymbolDataAbstract
+{
+public:
+  using value_type = T;
+
+  /**
+   * \brief A constructor.
+   *
+   * \param array_type_name specifying the name of the RAPID type of the contained items
+   * \param array_size specifying the size of the array to allocate
+   * \param value initiale value of all the item of the array
+   */
+  RAPIDArray(
+    const std::string& array_type_name, 
+    const size_t array_size,
+    const value_type& value = value_type()
+  )
+  :   array_type_name_ { array_type_name }
+  ,   container_ {std::vector<value_type>(array_size, value)}
+  {}
+
+  /**
+   * \brief A constructor that takes an array to initialize the underlying container
+   * 
+   * \param array_type_name specifying the name of the RAPID type of the contained items
+   * \param array Array that will be used to initialize the container
+   */
+  RAPIDArray(const std::string& array_type_name, const std::vector<value_type>&& array)
+  :   array_type_name_ {array_type_name}
+  ,   container_ {array}
+  {}
+
+  /**
+   * \brief A method for constructing a RAPID symbol data value string.
+   *
+   * \return std::string containing the constructed string.
+   */
+  std::string constructString() const override
+  {
+    std::stringstream ss;
+
+    ss << "[";
+
+    for (size_t i = 0; i < container_.size(); ++i)
+    {
+      ss << container_.at(i).constructString();
+
+      if (i != container_.size() - 1)
+      {
+        ss << ",";
+      }
+    }
+
+    ss << "]";
+
+    return ss.str();
+  }
+
+  /**
+   * \brief A method for parsing a RAPID symbol data value string.
+   *
+   * \param value_string containing the string to parse.
+   */
+  void parseString(const std::string& value_string) override
+  {
+    std::vector<std::string> substrings = extractDelimitedSubstrings(value_string);
+
+    if(container_.size() == substrings.size())
+    {
+      for (size_t i = 0; i < substrings.size(); ++i)
+      {
+        container_.at(i).parseString(substrings.at(i));
+      }
+    }
+  }
+
+  /**
+   * \brief A method for getting the type of item stored in the RAPID array.
+   *
+   * \return std::string containing the type.
+   */
+  std::string getType() const override
+  {
+    return array_type_name_;
+  }
+
+  /**
+   * \brief Operator for copying the RAPID array to another RAPID array.
+   *
+   * \param other containing the RAPID array to copy.
+   *
+   * \return RAPIDArray& containing the copy.
+   */
+  RAPIDArray<value_type>& operator=(const RAPIDArray<value_type>& other)
+  {
+    if (this != &other)
+    {
+      if (array_type_name_ == other.array_type_name_)
```

Comment:
The `other` has the same `value_type`. If the `array_type_name_` strings don't match, it is a logic error. Another reason to remove type string from the constructor arguments.